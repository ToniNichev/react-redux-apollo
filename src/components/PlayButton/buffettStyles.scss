@import '../../assets/stylesheets/utils/colors';
@import './styles';

.buffett.container {
  border-top: 5px solid transparent;
}

.base {
  .buffett & {
    background-color: $straw;
    font-size: 20px;
    transition: none;

    span {
      border: 0;
    }

    a:hover &,
    &:hover {
      background-color: $tawny-port;

      span {
        border: 0;
      }
    }
  }

  .buffett.container:hover & {
    background-color: $tawny-port;

    span {
      border: 0;
    }
  }
}

.featured {
  .buffett & {
    font-size: 18px;

    @include minWidth(tablet) {
      font-size: 32px;
    }
  }
}

.flyout {
  .buffett & {
    display: none;
  }
}
