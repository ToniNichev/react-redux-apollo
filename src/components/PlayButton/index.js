import React from 'react';
import PropTypes from 'prop-types';

let styles = require('./styles.scss');

/**
 * Pure React component function for displaying play button
 * @type {Function<Object>:React.Component}
 * @constructor
 * @augments {React.Component}
 * @param {bool} isFeatured Describes whether or not the playButton is
 * on featured media and therefore larger.
 * @param {string} type The type of playButton may be "play", "playlist",
 * @param {number} dataId Data id used for testing.
 * or "lock"
 */

const PlayButton = ({ isFeatured, type }) => {
  if (type.includes('buffett')) {
    const buffettStyles = require('./buffettStyles.scss'); // eslint-disable-line global-require

    styles = Object.assign(styles, buffettStyles);
  }

  const icons = {
    cnbc: 'icon-play-triangle',
    play: 'icon-play-triangle',
    playlist: 'icon-play-triangle',
    lock: 'icon-lock',
    'buffett-video': 'icon-buffett-video',
    'buffett-playlist': 'icon-buffett-playlist',
  };

  let wrapperStyles = styles.container;
  wrapperStyles = type.includes('buffett') ? `${styles.buffett} ${wrapperStyles}` : wrapperStyles;

  if (type === 'playlist') {
    return (
      <div className={wrapperStyles}>
        <div className={isFeatured ? styles.featured : styles.base} data-type={type}>
          <span className={styles.flyout}>{type === 'lock' ? 'log in' : 'watch now'}</span>
          <span className={`${icons[type]} ${styles.icon}`} />
        </div>
        <div className={isFeatured ? styles.featuredPlaylist : styles.playlist} />
      </div>
    );
  }

  return (
    <div className={wrapperStyles}>
      <div className={isFeatured ? styles.featured : styles.base} data-type={type}>
        <span className={styles.flyout}>{type === 'lock' ? 'log in' : 'watch now'}</span>
        <span className={`${icons[type]} ${styles.icon}`} />
      </div>
    </div>
  );
};

PlayButton.defaultProps = {
  isFeatured: false,
  type: 'play',
};

PlayButton.propTypes = {
  isFeatured: PropTypes.bool,
  type: PropTypes.string,
};

export default PlayButton;
