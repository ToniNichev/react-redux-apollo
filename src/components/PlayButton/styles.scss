@import '../../assets/stylesheets/utils/grid';
@import '../../assets/stylesheets/utils/colors';
@import '../../assets/stylesheets/utils/typography';
@import '../../assets/stylesheets/utils/mixins';

.container {
  @include fullyContained();
  position: absolute;
  border-top: 6px solid $gold;
  z-index: 1;
}

.base {
  background-color: $gold;
  color: $white;
  font-size: 12px;
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  top: -5px;
  right: 0;
  height: 40px;
  width: 40px;
  z-index: 1;
  transition: all 0.35s linear;

  @include minWidth(tablet) {
    font-size: 14px;
    height: 50px;
    width: 50px;
  }
}

.featured {
  composes: base;

  @include minWidth(tablet) {
    font-size: 22px;
    height: 80px;
    width: 80px;
  }
}

[data-type='playlist'] {
  box-shadow: inset 10 2px 10px #dcffa6, 10 2px 15px #000;
}

.icon {
  z-index: 2;
  display: block;
  height: 30px;
  width: 30px;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 2px solid $gold;
  transition: border 0.15s linear;

  a:hover &,
  a:focus &,
  button:hover &,
  button:focus &,
  .base:hover & {
    border: 2px solid $white;
  }

  @include minWidth(tablet) {
    height: 40px;
    width: 40px;

    .featured & {
      height: 60px;
      width: 60px;
      border: 2px solid $gold;
    }
  }
}

.flyout {
  @include fontWeight('Black');
  z-index: 1;
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  top: 0;
  right: 30px;
  height: 100%;
  width: 90px;
  background-color: $gold;
  font-size: 10px;
  letter-spacing: 1px;
  line-height: 1;
  text-align: center;
  text-transform: uppercase;
  color: rgba($dark-blue, 0);
  transform: scaleX(0);
  transform-origin: center right;
  transition: color 0.15s linear, transform 0.15s linear 0.15s;

  a:hover &,
  a:focus &,
  button:hover &,
  button:focus &,
  .base:hover & {
    transform: scaleX(1);
    color: $dark-blue;
    transition: color 0.15s linear 0.25s, transform 0.25s linear;
  }

  @include minWidth(tablet) {
    right: 50px;
    width: 130px;
    font-size: 12px;
    letter-spacing: 2.2px;

    .featured & {
      right: 80px;
      width: 200px;
      font-size: 20px;
    }
  }
}

.playlistWrapper {
  position: relative;
}

.playlist {
  position: absolute;
  top: 5px;
  right: 5px;
  height: 35px;
  width: 40px;
  background-color: $white;
  box-shadow: -5px 5px 0 0 $gold;

  a:hover &,
  a:focus &,
  button:hover &,
  button:focus &,
  &:hover {
    visibility: hidden;
  }

  @include minWidth(tablet) {
    height: 45px;
    width: 50px;
  }
}

.featuredPlaylist {
  composes: playlist;

  @include minWidth(tablet) {
    height: 75px;
    width: 80px;
  }
}
