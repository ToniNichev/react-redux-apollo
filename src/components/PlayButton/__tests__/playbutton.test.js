import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import PlayButton from '../index';

configure({ adapter: new Adapter() });

test('renders correctly', () => {
  const tree = renderer.create(<PlayButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
