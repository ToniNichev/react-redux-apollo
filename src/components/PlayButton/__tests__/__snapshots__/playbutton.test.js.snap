// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="base"
    data-type="play"
  >
    <span
      className="flyout"
    >
      watch now
    </span>
    <span
      className="icon-play-triangle icon"
    />
  </div>
</div>
`;
